Source/Moya.swift

```diff
@@ -449,8 +469,8 @@ internal extension MoyaProvider {
 public func convertResponseToResult(response: NSHTTPURLResponse?, data: NSData?, error: NSError?) ->
     Result<Moya.Response, Moya.Error> {
     switch (response, data, error) {
-    case let (.Some(response), .Some(data), .None):
-        let response = Moya.Response(statusCode: response.statusCode, data: data, response: response)
+    case let (.Some(response), data, .None):
+        let response = Moya.Response(statusCode: response.statusCode, data: data ?? NSData(), response: response)
```

Comment:
This makes sense to me, since "failing on empty data" was something Moya was doing differently from Alamofire, so this brings us more in line with their conventions.